### ISSUE TITLE
Changing the jinqJs repo for various reasons

### ISSUE THREAD

**Issue by mcoakley** [fordth/jinqJs#23]
I had an email discussion with @fordth about working on the structure of the jinqJs repo to make it more flexible for testing (including regression testing) as well as adding some build scripts. I've forked the repo to work on the issue I raised ( #21 ) and the fork repo is where I'll be doing the work.

If you are interested in letting me know your thoughts you can reply here or start a ticket within my [fork](https://github.com/mcoakley/jinqJs). 

Here are my goals:

- restructure the project folders for src and distribution
- implement testing structures to allow easy jinqJs testing for full coverage
  - testing of Node module
  - testing of Angular 1.x service
  - testing of TypeScript typings definition file
- implement a build process using TravisCI
- ensure repo is structured for easy development in a collaborative environment

Once done I'll submit the work done as a PR here.

**Comment by mcoakley** [comment]
In regard to structure I've removed the .idea and .settings folders as they are for IDE configurations. I added those to be ignored by git (in the .gitignore).

I would like to remove the versions folder as well but I could just exclude that through the .npmignore file so it isn't included in the NPM module. Thoughts?

**Comment by fordth** [comment]
Yes I agree, the versions folder isn't necessary. I first had it there for
a back reference of changes before I put the project in Git and just
continued updating it, but your correct it's no longer needed.
Thanks

**Comment by mcoakley** [comment]
I would like to remove the jinqjs-unstable.js file. Unstable versions can be handled with git branches. Additionally, I would like to move jinqjs.min.js to a build/dist folder and it would be created as part of the build process. If these are acceptable then I would like to follow a common convention I've noticed of putting the code into a lib folder. So we would have lib/jinqjs.js and no jinq.js variants in the repo root. Of course the package.json/etc would be updated to reflect these changes.

**Comment by fordth** [comment]
I agree, thanks
